File: lib/features/customers/domain/entities/customer.dart
class Customer {
  final int id;
  final String name;
  final String? companyName;
  final String? email;
  final String? phone;
  final String? address;
  final String type;

  Customer({
    required this.id,
    required this.name,
    this.companyName,
    this.email,
    this.phone,
    this.address,
    required this.type,
  });
}

File: backend/app/Models/Customer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Customer extends Model
{
    protected $fillable = [
        'name',
        'company_name',
        'email',
        'phone',
        'address',
        'type',
        'company_id',
    ];

    protected $casts = [
        'type' => 'string',
    ];

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }
}
